tests/parser_parity/harness.py

```
@@ -0,0 +1,229 @@
+
+
+def rust_parse(
+    query: str,
+    binary: Path,
+) -> tuple[dict[str, Any] | None, dict[str, Any] | None]:
+    """Run the Rust parse_json binary.
+
+    Returns ``(ast_dict, None)`` on success or ``(None, error_dict)`` on error.
+    """
+    try:
+        result = subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)